R/process_data.R

```diff
@@ -181,6 +186,73 @@ set_initial_plan_carsten_missings_to_zero <- function(data,
   return(data)
 }
 
+#' Remove rows from PACTA results that belong to company-sector combinations
+#' for which there is no positive production value in the relevant year of
+#' exposure (last year of forecast). This handles the edge case that a company
+#' may have a positive exposure for this sector, but none of the technologies
+#' covered in this analysis have any positive production. Such inconsistencies
+#' may arise e.g. because of unclear separation of the LDV and HDV sectors.
+#'
+#' @inheritParams calculate_annual_profits
+#' @inheritParams report_company_drops
+#' @param data tibble containing filtered PACTA results
+#'
+#' @return A tibble of data without rows with no exposure info
+#' @noRd
+remove_sectors_with_missing_production <- function(data,
+                                                   start_year,
+                                                   time_horizon,
+                                                   log_path) {
+  n_companies_pre <- length(unique(data$company_name))
+
+  # we merge the exposure of the last year of the forecast on the company
```

Comment:
1. why we merge?
2. the closest picture to me sounds a bit like this is only approximately correct, just making sure that the formulation is intended to imply that the solution may not always work?

Reply:
1. the comment is not required here, I kept it from another function by accident. Will remove. The explanation is in the function description
2. obsolete because of (1), but nonetheless: we filter on the exposure in the last year of the forecast because that exposure is likely the most similar to what an exposure in the shock year will look like. We cannot say exactly what that exposure will look like though, so yes, this is an approximation. But it will always technically work